controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -894,6 +894,13 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 		return actionUpdate{}
 	}
 
+	// Check if the host can support firmware components before creating the resrouce
```

Comment:
nit: resource

Reply:
not 100% sure if I should fix since this is a cherry-pick from an upstream commit.. .-.